src/backend/server.js

```diff
@@ -217,5 +321,121 @@ function createServer({ privateStorage, sharedStorage }) {
     }
   }
 
+  /**
+   * Create an kappa-osm database instance
+   *
+   * @param {object} options
+   * @param {string} options.indexDir Folder for storing leveldb and spatial indexes
+   * @param {string} options.feedsDir Folder for storing hypercore feeds
+   * @param {string} options.encryptionKey Encryption key for multifeed
+   * @returns {object} An object with properties `osm` (an instance of kappa-osm)
+   * and `close` (gracefully close all storage and callback)
+   */
+  function createOsmDb({ indexDir, feedsDir, encryptionKey }) {
+    const storages = [];
+
+    const indexDb = level(indexDir);
+    indexDb.on("error", err => {
+      main.bugsnag.notify(err, {
+        severity: "error",
+        context: "core"
+      });
+    });
+
+    const coreDb = kappa(createHypercoreStorage, {
+      valueEncoding: "json",
+      encryptionKey
+    });
+
+    // The main osm db for observations and map data
+    const osm = createOsmDbOrig({
+      core: coreDb,
+      index: indexDb,
+      storage: createBkdStorage
+    });
+
+    // To close cleanly we need to wait until replication has completed, destroy
+    // the discovery swarm, and close all the random-access-storage instances
+    // and the leveldb instance
+    const close = function close(cb) {
+      let pending = storages.length + 1;
```

Comment:
would this be storages.length + 2?

one for indexDb and the other for osm.core._logs.close?

Reply:
Thanks for the catch @okdistribute! I forgot to modify this when I added _logs.close. I've started to lean towards async/await for a lot of things, bugs like this and unhandled errors are too easy to miss with callback patterns.